### ISSUE TITLE
Fix the problem with using innerHTML

### ISSUE THREAD

**Issue by wataridori** [wataridori/chatpp#61]
Notice from **Mozilla Add-ons Editors**

> Please note the following for the next update:
> Using innerHTML to modify a document causes the entire document to be re-parsed, which is inefficient and has critical drawbacks, including invalidating any JavaScript reference to replaced DOM nodes, clearing any JavaScript properties and event listeners on replaced DOM nodes, and re-executing any script tags in the changed markup, and causing said scripts to fail if they rely on document.write.
https://developer.mozilla.org/en/XUL_School/DOM_Building_and_HTML_Insertion
> Additionally, due to both security and performance reasons (can be 100s of times slower), innerHTML (or similar method of converting strings to DOM e.g. outerHTML, insertAdjacentHTML, parseFromString, createContextualFragment, JQuery append/prepend/appendTo/html/before/after/insertBefore/insertAfter) should not be set using dynamic values. This can lead to security issues or fairly serious performance degradation.
> For inserting text, textContent (or JQuery text) or createTextNode() should be used instead of innerHTML.
> For inserting HTML, the safer method is to use createElement(), textContent, appendChild() instead of innerHTML.
> eg:
> js/extensions/contentscript.js#L167
> js/internals/all.js#L500 etc

> js/extensions/contentscript.js#L167

```
var text = "<li id=\"_chatppPreLoad\" style=\"display: inline-block\"><span id=\"chatppPreLoad\"></span></li>";
$("#_chatSendTool").append(text);
```

> js/internals/all.js#L425
```
var text = "<li id='_chatppSponsored' role='button' class=' _showDescription' aria-label='Chat Plus Plus Information'>" + ("<span id=\"chatppAdvertisement\" class=\"icoSizeSmall\">" + this.getAdvertisementText() + "</span>") + "</li>";
$("#_chatSendTool").append(text);
```

> Also: js/internals/all.js#L500, 509, 578, 611, 1419, 1555, 1557, 1665, 2097, 2274,2275, 2276, 2277, 2278

Have to fix these issues in the next Firefox update!

**Comment by wataridori** [comment]
Solution: 
- Refactor codes that contain jQuery's `append` function 
- Use `appendChild()` function instead 
Examples: https://stackoverflow.com/a/21498302

**Comment by wataridori** [comment]
Solved #69 and #70
